boundary.go

```diff
@@ -132,9 +178,8 @@ func (b *boundaryReader) isDelimiter(buf []byte) bool {
 
 	// Fast forward to the end of the boundary prefix
 	buf = buf[idx+len(b.prefix):]
-	buf = bytes.TrimLeft(buf, " \t")
 	if len(buf) > 0 {
-		if buf[0] == '\r' || buf[0] == '\n' {
+		if unicode.IsSpace(rune(buf[0])) {
```

Comment:
Do we have already have test coverage for this change?

Reply:
I am adding a test case for this change to TestBoundaryReader:
```
		{
			input:    "good\r\n--STOPHERE\t\r\nafter",
			boundary: "STOPHERE",
			want:     "good",
		},
```
Found use case in the wild.

Reply:
I added this test case